File: Mystreet.Domain/Entities/Order.cs
using Mystreet.Domain.Enums;

namespace Mystreet.Domain.Entities;

public class Order
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public string ShippingAddress { get; set; } = string.Empty;
    public string PaymentMethod { get; set; } = string.Empty;
    public decimal TotalAmount { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public List<OrderItem> Items { get; set; } = new();
}

File: frontend/mystreet-ui/src/types/order.ts
export type OrderItem = {
  productId: string;
  productName: string;
  size: string;
  quantity: number;
  unitPrice: number;
  imageUrl?: string | null;
};

export type Order = {
  id: string;
  status: number | string;
  totalAmount: number;
  createdAt: string;
  shippingAddress?: string;
  paymentMethod?: string;
  items?: OrderItem[];
};